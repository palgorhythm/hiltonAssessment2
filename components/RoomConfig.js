import React from 'react';

const RoomConfig = props => {
  const componentState = props.configState[props.roomNumber - 1];
  return (
    <div
      className={`${'room-container ' +
        (componentState.isActive ? 'active' : 'inactive')}`}>
      <h2>
        {props.roomNumber !== 1 ? (
          <input
            type="checkbox"
            checked={componentState.isActive}
            onChange={() => {
              props.handleCheckboxClick(
                props.roomNumber - 1,
                props.configState,
                props.setConfigState
              );
            }}
          />
        ) : (
          ''
        )}
        Room {props.roomNumber}
      </h2>
      <div className="room">
        <div className="dropdown-container">
          <h3>
            Adults
            <br />
            (18+)
          </h3>
          <select
            disabled={!componentState.isActive}
            value={componentState.numAdults}
            onChange={val => {
              props.handleSelect(val, props.roomNumber - 1, 'adults');
            }}>
            <option value="1">1</option>
            <option value="2">2</option>
          </select>
        </div>
        <div className="dropdown-container">
          <h3>
            Children
            <br />
            (0-17)
          </h3>
          <select
            disabled={!componentState.isActive}
            value={componentState.numChildren}
            onChange={val => {
              props.handleSelect(val, props.roomNumber - 1, 'children');
            }}>
            <option value="0">0</option>
            <option value="1">1</option>
            <option value="2">2</option>
          </select>
        </div>
      </div>
      <style jsx>{`
        :global(body) {
          margin: 0;
          font-family: -apple-system, BlinkMacSystemFont, Avenir Next, Avenir,
            Helvetica, sans-serif;
        }
        .room-container {
          display: flex;
          flex-direction: column;
          width: 200px;
          margin: 20px;
          background: lightgray;
          border-radius: 5px;
          border: 2px solid black;
          padding: 10px;
          box-shadow: 2px 1px 2px 1px rgba(120, 120, 120, 0.2);
        }

        .inactive {
          background: #aaa;
          border-color: gray;
        }

        .active .room {
          background: white;
        }
        .inactive .room {
          background: transparent;
        }
        .room {
          padding: 20px;
          display: flex;
          justify-content: space-around;
          border-radius: 5px;
        }
        h3 {
          margin: 0.5rem 0px;
        }
        input[type='checkbox'] {
          -webkit-appearance: none;
          width: 20px;
          height: 20px;
          margin: 0px 4px 0px 0px;
          background: white;
          border-radius: 5px;
          border: 2px solid #555;
        }
        input[type='checkbox']:checked {
          background: #a5eca1;
        }
        input[type='checkbox']:hover {
          cursor: pointer;
        }
        input[type='checkbox']:focus {
          outline: none;
        }
      `}</style>
    </div>
  );
};

export default RoomConfig;
