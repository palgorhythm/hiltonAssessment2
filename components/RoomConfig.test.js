import React from 'react';
import { shallow } from 'enzyme';
import RoomConfig from './RoomConfig';
describe('RoomConfig', () => {
  const initialMockState = [];
  for (let i = 0; i < 4; i += 1) {
    const isActive = i === 0 ? true : false;
    initialMockState.push({ isActive: isActive, numAdults: 1, numChildren: 0 });
  }

  const testProps = {
    configState: initialMockState,
    roomNumber: 1
  };
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<RoomConfig debug {...testProps} />);

    expect(component).toMatchSnapshot();
  });
});
