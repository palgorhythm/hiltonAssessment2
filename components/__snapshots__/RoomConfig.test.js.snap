// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomConfig should render correctly in "debug" mode 1`] = `
<div
  className="jsx-1890337735 room-container active"
>
  <h2
    className="jsx-1890337735"
  >
    Room 
    1
  </h2>
  <div
    className="jsx-1890337735 room"
  >
    <div
      className="jsx-1890337735 dropdown-container"
    >
      <h3
        className="jsx-1890337735"
      >
        Adults
        <br
          className="jsx-1890337735"
        />
        (18+)
      </h3>
      <select
        className="jsx-1890337735"
        disabled={false}
        onChange={[Function]}
        value={1}
      >
        <option
          className="jsx-1890337735"
          value="1"
        >
          1
        </option>
        <option
          className="jsx-1890337735"
          value="2"
        >
          2
        </option>
      </select>
    </div>
    <div
      className="jsx-1890337735 dropdown-container"
    >
      <h3
        className="jsx-1890337735"
      >
        Children
        <br
          className="jsx-1890337735"
        />
        (0-17)
      </h3>
      <select
        className="jsx-1890337735"
        disabled={false}
        onChange={[Function]}
        value={0}
      >
        <option
          className="jsx-1890337735"
          value="0"
        >
          0
        </option>
        <option
          className="jsx-1890337735"
          value="1"
        >
          1
        </option>
        <option
          className="jsx-1890337735"
          value="2"
        >
          2
        </option>
      </select>
    </div>
  </div>
  <JSXStyle
    id="1890337735"
  >
    body{margin:0;font-family:-apple-system,BlinkMacSystemFont,Avenir Next,Avenir, Helvetica,sans-serif;}.room-container.jsx-1890337735{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;width:200px;margin:20px;background:lightgray;border-radius:5px;border:2px solid black;padding:10px;box-shadow:2px 1px 2px 1px rgba(120,120,120,0.2);}.inactive.jsx-1890337735{background:#aaa;border-color:gray;}.active.jsx-1890337735 .room.jsx-1890337735{background:white;}.inactive.jsx-1890337735 .room.jsx-1890337735{background:transparent;}.room.jsx-1890337735{padding:20px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:space-around;-webkit-justify-content:space-around;-ms-flex-pack:space-around;justify-content:space-around;border-radius:5px;}h3.jsx-1890337735{margin:0.5rem 0px;}input[type='checkbox'].jsx-1890337735{-webkit-appearance:none;width:20px;height:20px;margin:0px 4px 0px 0px;background:white;border-radius:5px;border:2px solid #555;}input[type='checkbox'].jsx-1890337735:checked{background:#a5eca1;}input[type='checkbox'].jsx-1890337735:hover{cursor:pointer;}input[type='checkbox'].jsx-1890337735:focus{outline:none;}
  </JSXStyle>
</div>
`;
